src/NuGet.Core/NuGet.Common/MachineCache.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace NuGet.Common
+{
+    /// <summary>
+    /// The machine cache represents a location on the machine where packages are cached. It is a specific implementation of a local repository and can be used as such.
+    /// NOTE: this is a shared location, and as such all IO operations need to be properly serialized
+    /// </summary>
+    public class MachineCache
+    {
+        private const string NuGetCachePathEnvironmentVariable = "NuGetCachePath";
+        /// <summary>
+        /// Determines the cache path to use for NuGet.exe. By default, NuGet caches files under %LocalAppData%\NuGet\Cache.
+        /// This path can be overridden by specifying a value in the NuGetCachePath environment variable.
+        /// </summary>
+        public static string GetCachePath()
+        {
+#if IS_CORECLR
+            string localAppDataPath = NuGetEnvironment.GetFolderPath(NuGetFolderPath.HttpCacheDirectory);
+            return GetCachePath(Environment.GetEnvironmentVariable, localAppDataPath);
+#else
+            string localAppDataPath = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            return GetCachePath(Environment.GetEnvironmentVariable, localAppDataPath);
+#endif
+        }
+
+        public static string GetCachePath(Func<string, string> getEnvironmentVariable, string localAppDataPath)
+        {
+            string cacheOverride = getEnvironmentVariable(NuGetCachePathEnvironmentVariable);
+            if (!String.IsNullOrEmpty(cacheOverride))
```

Comment:
[type name](https://github.com/NuGet/Home/wiki/Coding-Guidelines#use-c-type-keywords-in-favor-of-net-type-names)